### PR TITLE
OCPQE-16679: libvirt deployment

### PR DESCRIPTION
**Pull request by zniu1011**

### CODE REVIEW

File: images/fcos-auxiliary-image/root/usr/bin/libvirt_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    if [ "$status" = "disabled" ]; then
+        systemctl enable libvirtd
+    fi
+fi
+# Check the 'default' pool existing, define the pool if it doesn't exist
+default_image_pool=`virsh pool-list --all | grep default`
+
+if [[ -z ${default_image_pool} ]]; then
+    cat > "/etc/libvirt/storage/default.xml" << EOF
+    <pool type='dir'>
```
Comment (aleskandro) [inline images/fcos-auxiliary-image/root/usr/bin/libvirt_deploy.sh:25]: I'd remove the indentation here

File: images/fcos-auxiliary-image/root/usr/bin/libvirt_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    is_active=`systemctl is-active libvirtd`
+    if [ "$status" = "inactive" ]; then
+        systemctl start libvirtd
+    fi
+# Enable libvirtd if it is disabled
+    is_enabled=`systemctl is-enabled libvirtd`
+    if [ "$status" = "disabled" ]; then
+        systemctl enable libvirtd
+    fi
+fi
```
Comment (aleskandro) [inline images/fcos-auxiliary-image/root/usr/bin/libvirt_deploy.sh:19]: we do not need all this snipped as we can ensure that via the layered image and by setting libvirt enabled by default.

**Review by aleskandro — COMMENTED** [review]
Hi @zniu1011 , thanks! I added some comments to the current script.

Moreover, we need this script to run as a systemd unit, "after" the `libvirtd.service` unit starts and "required" by `libvirtd.service`. 

We also need to setup libvirtd as an enabled by default service, which means having a file that lists it as in https://github.com/openshift-qe/baremetal-qe-infra/blob/master/images/fcos-base-image/root/usr/lib/systemd/system-preset/97-fedora-custom.preset

Therefore, we need to:
1. Add a unit file for this  script, to execute after libvirtd.service starts
2. We need to add libvirtd.service in the `root/usr/lib/systemd/system-preset/97-auxiliary.preset`  (create the path)
3. The dockfile for the auxiliary image should change to copy the `root/` content and to run `systemctl preset-all`, similarly to this line for the base image https://github.com/openshift-qe/baremetal-qe-infra/blob/master/images/fcos-base-image/Containerfile#L71

**Comment by zniu1011** [comment]
Thanks @aleskandro, I created the service file, and updated.

**Review by aleskandro — COMMENTED** [review]
Hi @zniu1011 just a few nits and a less minor issue that may be a fault of mine in describing the task initially.

This work will be needed also in the bastion host, so we need to move these changes to the base image.

**Comment by zniu1011** [comment]
Thanks @aleskandro, I restore the auxiliary server, and move it to base image. Please check if there is further issue.

**Pull request merged**